packages/examples/src/scenes/components-switch.tsx

```diff
@@ -0,0 +1,85 @@
+import {Circle, Node, NodeProps, Rect} from '@motion-canvas/2d/lib/components';
+import {easeInOutCubic, tween} from '@motion-canvas/core/lib/tweening';
+import {
+  Color,
+  ColorSignal,
+  PossibleColor,
+} from '@motion-canvas/core/lib/types/Color';
+import {initial, signal} from '@motion-canvas/2d/lib/decorators';
+import {
+  createSignal,
+  SignalValue,
+  SimpleSignal,
+} from '@motion-canvas/core/lib/signals';
+import {createRef} from '@motion-canvas/core/lib/utils';
+import {all} from '@motion-canvas/core/lib/flow';
+
+export interface SwitchProps extends NodeProps {
+  initialState?: SignalValue<boolean>;
+  accent?: SignalValue<PossibleColor>;
+}
+
+export class Switch extends Node {
+  @initial(false)
+  @signal()
+  public declare readonly initialState: SimpleSignal<boolean, this>;
+
+  @initial('#68ABDF')
+  @signal()
+  public declare readonly accent: ColorSignal<this>;
+
+  private isOn: boolean;
+  private readonly indicatorPosition = createSignal(0);
+  private readonly offColor = new Color('#242424');
+  private readonly indicator = createRef<Circle>();
+  private readonly container = createRef<Rect>();
+
+  public constructor(props?: SwitchProps) {
+    super({
+      ...props,
+    });
+
+    this.indicatorPosition(this.initialState ? 50 : -50);
+    this.isOn = this.initialState();
+
+    this.add(
+      <>
```

Comment:
because if you were to make a layout component and wanted to ensure it was a layout, you would use :
```tsx
super({
   layout
   ...props
})
```
just makes it a bit more obvious not only props can go here

Reply:
this is true, it was 11 and it slipped past me lol. will be fixed